src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ModuleBuilderImpl.cs

```diff
@@ -557,10 +582,10 @@ private void AddGenericTypeParametersAndConstraintsCustomAttributes(EntityHandle
                 bodyOffset: offset,
                 parameterList: MetadataTokens.ParameterHandle(parameterToken));
 
-        private TypeReferenceHandle AddTypeReference(Type type, AssemblyReferenceHandle parent) =>
+        private TypeReferenceHandle AddTypeReference(Type type, EntityHandle resolutionScope) =>
             _metadataBuilder.AddTypeReference(
-                resolutionScope: parent,
-                @namespace: (type.Namespace == null) ? default : _metadataBuilder.GetOrAddString(type.Namespace),
+                resolutionScope: resolutionScope,
+                @namespace: (type.Namespace == null || type.IsNested) ? default : _metadataBuilder.GetOrAddString(type.Namespace),
```

Comment:
Why is the `type.IsNested` check needed?

Reply:
Right, it's not needed.